### ISSUE TITLE
Helm Charts - Separate image into repository and tag

### ISSUE THREAD

**Issue by krishau99** [rook/rook#13585]
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**

The feature should separate the helm value "image" into "repository" and "tag" in both the Rook Ceph Operator and Cluster charts. 

**What is use case behind this feature:**

As I am using my own private image registry, I need to modify the image repository in the values of my helm-chart. Since the tag is also specified in this same section, this makes it impossible for me to go with the default tags in case of an upgrade. I would then need to manually update the tags of every image for every upgrade. Separating "image" into "repository" and "tag" would make it much easier for people using a private registry to apply new versions.

**Environment**:

Any environment using a private image registry.

**Comment by travisn** [comment]
Ah, the csi images.  It would keep things simpler the way defaults are handled for the csi images, but I don't see anything preventing them from being separated either. @Madhu-1 ?

**Comment by Madhu-1** [comment]
> Ah, the csi images. It would keep things simpler the way defaults are handled for the csi images, but I don't see anything preventing them from being separated either. @Madhu-1 ?

Yes correct we can do it, i would also suggest making ImagePullPolicy as also configurable for csi images as well :)

**Comment by davralin** [comment]
@travisn - right, that makes sense.

How ofte should I expect changes in that default value then, every helm release, or every major?

**Comment by travisn** [comment]
> @travisn - right, that makes sense.
> 
> How ofte should I expect changes in that default value then, every helm release, or every major?

The default value is updated whenever a new Ceph version is released for the latest major supported version of Ceph, which lately is every few months. Currently it is v19.2.0 for the past few months and v19.2.1 is due anytime now.
